### PR TITLE
RFC: modify buildsystem docs in README.md

### PR DESCRIPTION
**Pull request by akira25**
This is meant to be an first improvement for the issues in documentation addressed in #824. This commit mainly takes some of documentation of weimarnetz and adds that to our README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,6 +51,34 @@ pacman -S base-devel git ncurses lib32-zlib gawk time unzip perl-xml-libxml \
...
 
+
+### Build-System-Structure  
+
+Where can I change something? What does these files do? 
+
+```
+:: Settings
+
+config.mk             - OpenWRT git repository, revision, select what package lists to build
```
Comment (SvenRoederer) [inline README.md:62]: the OpenWrt-Git stuff has been moved to the _modules_ file.

File: README.md [hunk 2]
```diff
@@ -48,6 +51,34 @@ pacman -S base-devel git ncurses lib32-zlib gawk time unzip perl-xml-libxml \
...
+feeds.conf            - OpenWRT feeds.conf that is used for the package feeds during build
```
Comment (SvenRoederer) [inline README.md:63]: The whole file moved to the _modules_ file

**Comment by SvenRoederer** [comment]
Improved docs are always welcome.

Before putting much work into this, we should discuss on https://github.com/freifunk-berlin/firmware/pull/765 as it will change many of the existing folders.

File: README.md (hunk 2 above)
Comment (Akira25) [reply README.md]: adjusted in new commit

File: README.md (hunk 1 above)
Comment (Akira25) [reply README.md]: adjusted in new version.

**Comment by Akira25** [comment]
> Improved docs are always welcome.
> 
> Before putting much work into this, we should discuss on #765 as it will change many of the existing folders.

I'm already finished. I'd prefer to have this merged fairly soon, as anyone how like to use our buildsystem *now*, would need to invest much debugging effort into it, until he gets to know, where to configure anything.

**Comment by Akira25** [comment]
Besides that, regarding our current development velocity, I don't think we will have a decision on #765 within the next 1.5 Years.

File: README.md [hunk 3]
```diff
@@ -14,15 +15,17 @@ More user relevant information about the firmware are on the wiki page at: https
...
-For the Berlin Freifunk firmware we use vanilla OpenWrt with additional patches
-and packages. The Makefile automates firmware
-creation and apply patches / integrates custom freifunk packages. All custom
-patches are located in *patches/* and all additional packages can be found at
-http://github.com/freifunk-berlin/packages_berlin.
+How does this stuff work? 
+
+The idea is to download OpenWRT via git, patch it with all patches in the `patches/` folder and configure it according to the configuration snippets in `configs`. 
+Then use the OpenWRT buildsystem to build all packages and the Imagebuilder and use the latter to assemble the final image with lists of packages in packages for the router.
+All additional packages can be found at http://github.com/freifunk-berlin/packages_berlin.
```
Comment (SvenRoederer) [inline README.md:27]: I'm not sure, if there is some rule about line-breaks in the source, as it was in the above code before your change.

File: README.md [hunk 4]
```diff
@@ -48,6 +51,34 @@ pacman -S base-devel git ncurses lib32-zlib gawk time unzip perl-xml-libxml \
...
+```
+:: Settings
+
+config.mk             - select what package lists to build
+modules               - define git repository, revision, etc of OpenWRT and Packages
+
+patches/              - Patches against the OpenWRT directory 
+packagelists/         - Package Lists 
+profiles/             - List of router profiles for each architecture - names come from the Imagebuilder
+
```
Comment (SvenRoederer) [inline README.md:68]: probably you like to mention the "embedded-files" also, see: https://github.com/freifunk-berlin/firmware/commit/a66896d3ca02d1aff22be0c5b77e812ebd7596d4

**Review by SvenRoederer — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (foudfou) [inline README.md:27]: Shouldn't this be https://github.com/freifunk-berlin/firmware-packages ?

**Review by foudfou — COMMENTED** [review]
Very helpful!

File: README.md (hunk 3 above)
Comment (Akira25) [reply README.md:27]: Right. That is a relict from the weimar-netz text.

File: README.md [hunk 5]
```diff
@@ -129,6 +162,16 @@ additional options
   * "yes" the Makefile will always act as "IS_BUILDBOT" was set to "yes"
   * "no"  the Makefile will always act as "IS_BUILDBOT" was set to "no" / is unset. This way we can run builds on the buildbot like a local build.
 
+
+### Build a new router model for an existing architecture
+
+Everything that is in the [Table of Hardware](https://openwrt.org/toh/start) should work at least in theory. Wifi should be ath9k / ath10k. mt76 also appears to work. 
+
+You can use the Imagebuilder to get a list of all supported models for a platform. You can also look in the [source](https://git.openwrt.org/?p=openwrt/openwrt.git;a=tree;f=target/linux/ar71xx/image) for model names.
```
[the remark this answers is not in the document]
Comment (Akira25) [reply README.md:170]: Jep. I'll use ath79. Will be included into next commit.

File: README.md [hunk 6]
```diff
@@ -129,6 +162,16 @@ additional options
...
   * "no"  the Makefile will always act as "IS_BUILDBOT" was set to "no" / is unset. This way we can run builds on the buildbot like a local build.
 
+
+### Build a new router model for an existing architecture
+
+Everything that is in the [Table of Hardware](https://openwrt.org/toh/start) should work at least in theory. Wifi should be ath9k / ath10k. mt76 also appears to work. 
+
+You can use the Imagebuilder to get a list of all supported models for a platform. You can also look in the [source](https://git.openwrt.org/?p=openwrt/openwrt.git;a=tree;f=target/linux/ar71xx/image) for model names. 
+
+Just put the name in the `/profiles/<target>.profiles` and it should work.
```
[the remark this answers is not in the document]
Comment (Akira25) [reply README.md:172]: right. I added that.

File: README.md [hunk 7]
```diff
@@ -14,15 +15,17 @@ More user relevant information about the firmware are on the wiki page at: https
...
 ### Info
 
-For the Berlin Freifunk firmware we use vanilla OpenWrt with additional patches
-and packages. The Makefile automates firmware
-creation and apply patches / integrates custom freifunk packages. All custom
-patches are located in *patches/* and all additional packages can be found at
-http://github.com/freifunk-berlin/packages_berlin.
+How does this stuff work? 
+
+The idea is to download OpenWRT via git, patch it with all patches in the `patches/` folder and configure it according to the configuration snippets in `configs`.
```
[the remark this answers is not in the document]
Comment (Akira25) [reply README.md:25]: right. That wasn't quite precise, sorry. Is included into next commit.

File: README.md (hunk 3 above)
Comment (Akira25) [reply README.md:27]: Im not sure either. When I opened that in my editor, it looked fairly strange, so I changed it. For the parser at github that shouldn't matter anyway. But maybe we should consider, that this README.md might be opened in an editor on a terminal.

Do you think it would be a good idea, to assume a 80-char wide terminal?

File: README.md (hunk 4 above)
Comment (Akira25) [reply README.md:68]: Nice feature. Do we use it in current builds? I did not see the directory `$(FW_DIR)/embedded-files` in the current master branch. Is it constructed at build-time?

File: README.md [hunk 8]
```diff
@@ -14,15 +15,17 @@ More user relevant information about the firmware are on the wiki page at: https
...
-For the Berlin Freifunk firmware we use vanilla OpenWrt with additional patches
-and packages. The Makefile automates firmware
-creation and apply patches / integrates custom freifunk packages. All custom
-patches are located in *patches/* and all additional packages can be found at
-http://github.com/freifunk-berlin/packages_berlin.
+How does this stuff work? 
+
+The idea is to download OpenWrt via git, patch it with all patches in the `patches/` folder and configure it according to the configuration snippets in `configs/`. 
+Then use the OpenWrt buildsystem to build all packages and the Imagebuilder and use the latter to assemble the final image with lists of packages in packages for the router.
+All additional packages can be found at https://github.com/freifunk-berlin/firmware-packages.
```
Comment (SvenRoederer) [inline README.md:27]: Just read the sentence .... Probably better to write "All Berlin.Freifunk specific packages can be found...".
As we currently include the Gluon-packages feed and the freifunk-packages feed. Probably other feeds (e.g. freifunk-Potsdam) will also be present one time.

File: README.md [hunk 9]
```diff
@@ -48,6 +51,34 @@ pacman -S base-devel git ncurses lib32-zlib gawk time unzip perl-xml-libxml \
  flex wget gettext quilt python2 openssl
 ```
 
+
+### Build-System-Structure  
+
+Where can I change something? What does these files do?
```
Comment (SvenRoederer) [inline README.md:57]: isn't is "What do these files do" or "What does a file do"?

File: README.md (hunk 4 above)
Comment (SvenRoederer) [inline README.md:68]: Why not mentioning "configs/" here?

File: README.md [hunk 10]
```diff
@@ -129,6 +162,16 @@ additional options
...
 
+
+### Build a new router model for an existing architecture
+
+Everything that is in the [Table of Hardware](https://openwrt.org/toh/start) should work at least in theory. Wifi should be ath9k / ath10k. mt76 also appears to work. 
+
+You can use the Imagebuilder to get a list of all supported models for a platform. You can also look in the [source](https://git.openwrt.org/?p=openwrt/openwrt.git;a=tree;f=target/linux/ath79/image) for model names. 
+
+Just put the name in the `/profiles/<target>.profiles` and it should work. Assure, that there is also a matching configuration for that target at `/configs/`
+
```
Comment (SvenRoederer) [inline README.md:173]: Unify the use of target vs. platform. In line 170 you write "all supported models for a platform", in line 172 "matching configuration for that target ".
I prefer using the target term, as it's also "upstream" speak.

File: README.md [hunk 11]
```diff
@@ -150,18 +194,20 @@ For inspiration please take a look at the related
 
 For a new release, create a new branch. The branch name must be a semantic version
 number. Make sure you change the semantic version number and, for major releases,
-the codename in the README and config files (./configs/*)
+the codename in the README and config files (`./configs/`)
```
Comment (SvenRoederer) [inline README.md:197]: drop the "./" prefix here, as you did everywhere in the document.

Comment (SvenRoederer) [reply README.md:197]: by the way: currently you can directly reference "configs/common.config" for changing the codename.

File: README.md (hunk 10 above)
Comment (SvenRoederer) [reply README.md:173]: also remove the leading "/" for the folders.

File: README.md (hunk 3 above)
Comment (SvenRoederer) [reply README.md]: 80 chars seem fine.

File: README.md (hunk 4 above)
Comment (SvenRoederer) [reply README.md:68]: * https://git.wiki.kernel.org/index.php/GitFaq#Can_I_add_empty_directories.3F
I personally use it to include my ssh-public key in my test-builds (https://github.com/SvenRoederer/freifunk-berlin-firmware/tree/master/embedded-files/etc/dropbear), but I would not use is that much. 
There was an approach to use it for the banner, but this did not work as expected (https://github.com/freifunk-berlin/firmware/commit/c2573c99731aa9649327160c617e53d31e22da3d). Currently PR https://github.com/freifunk-berlin/firmware/pull/792 makes some use of it.

Comment (Akira25) [reply README.md:68]: Done.

Comment (Akira25) [reply README.md:68]: What does the files in `configs/` exactly do? It looks like they handle some kernel and package-foo...

File: README.md (hunk 10 above)
Comment (Akira25) [reply README.md:173]: Okay, both done.

File: README.md (hunk 8 above)
Comment (Akira25) [reply README.md:27]: added.

File: README.md (hunk 9 above)
Comment (Akira25) [reply README.md:57]: right. Wasn't quite sure on that one.

File: README.md (hunk 11 above)
Comment (Akira25) [reply README.md:197]: Done.

File: README.md (hunk 4 above)
Comment (SvenRoederer) [reply README.md:80]: common.config: contains the OpenWrt config-options for all targets
target.config: gets added to the common.config and contains the target-specific config-options. possibly overwrites earlier settings.
common-autobuild.config: gets added on top , if IS_BUILDBOT=yes is defined

see: https://github.com/freifunk-berlin/firmware/blob/master/Makefile#L122 and following lines

File: README.md [hunk 12]
```diff
@@ -1,250 +1,326 @@
...
-broken stuff in OpenWrt itself or for example LuCI) and additional default packages/configuration settings.
 
 ## Contact / More information
 
-More user relevant information about the firmware are on the wiki page at: https://wiki.freifunk.net/Berlin:Firmware. There you can also find the
-* [ReleaseNotes](https://wiki.freifunk.net/Berlin:Firmware/v1.0.2)
-* a tutorial ([en](https://wiki.freifunk.net/Berlin:Firmware:En:Howto) / [de](https://wiki.freifunk.net/Berlin:Firmware:Howto)) on router configuration
+More user relevant information on the firmware are on the wiki page at:
+https://wiki.freifunk.net/Berlin:Firmware. There you can also find the 
+* [ReleaseNotes](https://wiki.freifunk.net/Berlin:Firmware/v1.0.2) and
```
Comment (SvenRoederer) [inline README.md:13]: Hedy-1.0.6 is out and should be referenced here. But this can be a separate commit.

File: README.md [hunk 13]
```diff
@@ -1,250 +1,326 @@
...
+``` 
+:: Settings
+
+config.mk         - select what package lists to build
+configs/          - 
+modules           - define git repository, revision, etc of OpenWRT and Packages
+
+patches/          - Patches against the OpenWRT directory  
+packagelists/     - Package Lists  
+profiles/         - List of router profiles for each architecture - names come
```
Comment (SvenRoederer) [inline README.md:75]: probably use "target architecture" to keep the common target term

File: README.md [hunk 14]
```diff
@@ -1,250 +1,326 @@
...
+:: Settings
+
+config.mk         - select what package lists to build
+configs/          - 
+modules           - define git repository, revision, etc of OpenWRT and Packages
+
+patches/          - Patches against the OpenWRT directory  
+packagelists/     - Package Lists  
+profiles/         - List of router profiles for each architecture - names come 
+                      from the Imagebuilder
```
Comment (SvenRoederer) [inline README.md:76]: is it worth to mention "make info" here?

File: README.md [hunk 15]
```diff
@@ -1,250 +1,326 @@
...
+embedded-files/   - Files that will be included in the images without further
+                      processing. Its mirrored to / . Thus, you must provide a
+                      directory tree there, to put your file at the right point
+
+:: Building 
+
+Makefile                - Does all the stuff. Cloning and Updating OpenWRT,
```
Comment (SvenRoederer) [inline README.md:83]: * use updating in lower-case
* it's OpenWrt

File: README.md [hunk 16]
```diff
@@ -1,250 +1,326 @@
...
 
+### Directory Layout of `firmwares/`
+
+You can find the actual firmware images generated by the ImageBuilder (and the
+ImageBuilder itself) in `firmwares/`. The layout looks like the following:
+
+``` 
+firmwares/ 
+  TARGET/ 
+    backbone/
```
Comment (SvenRoederer) [inline README.md:123]: Should we use the changed packagelists here (notunnel, manual, ...)?

File: README.md [hunk 17]
```diff
@@ -1,250 +1,326 @@
...
 
-As you notice there are several different image variants ("backbone", "default", etc.).
-These different *packages lists* are defined in `packages/`.
-See the "Features" section above for a description of the purpose of each package list.
-With the "OpenWrt-Imagebuilder" you can assemble your own image variant with your
-*packages lists* without having to compile everything yourself. The "OpenWrt-SDK" is
-the fastest way to build your own packages or programs without compiling OpenWrt itself.
-The "initrd" directory contains some initrd-images for netboot, which are required on
-some boards to initially install OpenWrt.
+As you notice there are several different image variants ("backbone", "default",
```
Comment (SvenRoederer) [inline README.md:143]: See above: Should we use the changed packagelists here (notunnel, manual, ...)?

File: README.md [hunk 18]
```diff
@@ -1,250 +1,326 @@
...
 ### customizing make
 
 `make` will use by default `TARGET` and `PACKAGES_LIST_DEFAULT` defined in
 `config.mk`. You can customize this by overriding them:
 
-```
-make TARGET=mpc85xx PACKAGES_LIST_DEFAULT=backbone
-```
-in addition you can build your own image from a prebuilt imagebuilder by something like:
+``` make TARGET=mpc85xx PACKAGES_LIST_DEFAULT=backbone ```
```
Comment (SvenRoederer) [inline README.md:161]: As you touch it: 
* Since some time it's mpc85xx-generic
* Should we use the changed packagelists here (notunnel, manual, ...)?

File: README.md [hunk 19]
```diff
@@ -1,250 +1,326 @@
...
 
-The default target is `ar71xx-generic`. For a complete list of supported targets look in `configs/` for the target-specific configs.
-Each of these targets need a matching file in `profiles/` with the profiles (boards) that should be build with the imagebuilder.
+``` make images IB_FILE=<file> TARGET=... PACKAGES_LIST_DEFAULT=... ```
 
-additional options
+The default target is `ar71xx-generic`. For a complete list of supported targets
+look at `configs/` for the target-specific configs. Each of these targets need a
+matching file in `profiles/` with the profiles (boards) that should be build
+with the imagebuilder.
```
Comment (SvenRoederer) [inline README.md:171]: "should be build **by** the imagebuilder"?

File: README.md [hunk 20]
```diff
@@ -1,250 +1,326 @@
...
+      unset. This way we can run builds on the buildbot like a local build.
+
+
+### Build a new router model for an existing architecture
+
+Everything that is in the [Table of Hardware](https://openwrt.org/toh/start)
+should work at least in theory. Wifi should be ath9k / ath10k. mt76 also appears
+to work. 
+
+You can use the Imagebuilder to get a list of all supported models for a
```
Comment (SvenRoederer) [inline README.md:195]: Probably the better place, than the mentioned above,  to reference "make info" if we want to reference it at all.

File: README.md [hunk 21]
```diff
@@ -1,250 +1,326 @@
...
+the "stable" downloads directory:
 [http://buildbot.berlin.freifunk.net/buildbot/stable/](http://buildbot.berlin.freifunk.net/buildbot/stable/)
 
-All branches with names not fitting the "X.Y.Z" pattern are built and put into the "unstable" directory:
+All branches with names not fitting the "X.Y.Z" pattern are built and put into
+the "unstable" directory:
 [http://buildbot.berlin.freifunk.net/buildbot/unstable/](http://buildbot.berlin.freifunk.net/buildbot/unstable/)
-Note that in the directory there is no reference to the branch name; unstable builds can be identified by build number only.
+Note that in the directory there is no reference to the branch name; unstable
+builds can be identified by their build number only.
```
Comment (SvenRoederer) [inline README.md:222]: Are you willing to referernce the VERSION.txt file here

File: README.md [hunk 22]
```diff
@@ -1,250 +1,326 @@
...
 
+
 ### Patches with "git format-patch"
 
 **Important:** all patches should be pushed upstream!
 
-If a patch is not yet included upstream, it can be placed in the corresponding subdirectory below the`patches`
-directory. To create a correct patch-file just use the [`git format-patch`](https://git-scm.com/docs/git-format-patch) command.
+If a patch is not yet included upstream, it can be placed in the corresponding
+subdirectory below the `patches` directory. To create a correct patch-file just
```
Comment (SvenRoederer) [inline README.md:248]: patches with "/" suffix?

File: README.md [hunk 23]
```diff
@@ -1,250 +1,326 @@
...
 
-```
-git format-patch origin
-```
+``` git format-patch origin ```
 
 Send a patch to the OpenWrt mailing list with `git send-email`:
 
-```
-git send-email \
```
Comment (SvenRoederer) [inline README.md:242]: why removing the indention?

File: README.md [hunk 24]
```diff
@@ -1,250 +1,326 @@
...
+If a patch is not yet included upstream, it can be placed in the corresponding
+subdirectory below the `patches` directory. To create a correct patch-file just
+use the [`git format-patch`](https://git-scm.com/docs/git-format-patch) command.
+
 
 #### Create a patch
 
 In order to add a patch file update your build environment by running:
 
-```bash
```
Comment (SvenRoederer) [inline README.md:170]: on all the following code-blocks you broke the layout. please compare with the current file.

File: README.md (hunk 12 above)
Comment (pmelange) [reply README.md:13]: Is it possible to use a variable which accesses the latest tagged release version.  (for the separate commit which will someday materialize)

File: README.md (hunk 19 above)
Comment (pmelange) [reply README.md:171]: "sould be built by" would be the queen's english.

File: README.md (hunk 12 above)
Comment (SvenRoederer) [reply README.md:13]: As the Readme belongs to the Codebase of a release, releasing something with an old or template value here, will end in having this template in the released Readme. Updating it after releasing creates a new version of the file (commit) which should cause  to release again. So the never ending loop has started ...

In long-term I suggest to get rid of the release-history in the Readme of the firmware-buildsystem, like I did here: https://github.com/freifunk-berlin/firmware/tree/745bda009b313cf3ce4eaacd82e232a4644f8a03

File: README.md (hunk 4 above)
Comment (Akira25) [reply README.md:75]: added in current version.

File: README.md (hunk 12 above)
Comment (Akira25) [reply README.md]: As the page for v1.0.6 does not exist, I linked the page for v1.0.3

File: README.md (hunk 15 above)
Comment (Akira25) [reply README.md]: Done with regular expression. Should be fine now.

File: README.md (hunk 16 above)
Comment (Akira25) [reply README.md]: Adjusted to new naming.

File: README.md (hunk 18 above)
Comment (Akira25) [reply README.md]: Adjusted to new naming.

File: README.md (hunk 23 above)
Comment (Akira25) [reply README.md]: reversed.

File: README.md (hunk 24 above)
Comment (Akira25) [reply README.md]: reversed.

File: README.md (hunk 22 above)
Comment (Akira25) [reply README.md]: adjusted.

File: README.md (hunk 17 above)
Comment (Akira25) [reply README.md]: included.

File: README.md [hunk 25]
```diff
@@ -1,28 +1,36 @@
...
-For the Berlin Freifunk firmware we use vanilla OpenWrt with additional patches
-and packages. The Makefile automates firmware
-creation and apply patches / integrates custom freifunk packages. All custom
-patches are located in *patches/* and all additional packages can be found at
-http://github.com/freifunk-berlin/packages_berlin.
+How does this stuff work? 
+
+The idea is to download OpenWrt via git, patch it with all patches in the
+`patches/` folder and configure it according to the configuration snippets in
+`configs`.  Then use the OpenWrt buildsystem to build all packages and the
```
Comment (SvenRoederer) [inline README.md:29]: should be configs/ , as it's a folder

File: README.md [hunk 26]
```diff
@@ -1,28 +1,36 @@
...
+Imagebuilder and use the latter to assemble the final image with lists of
+packages in packages for the router.
```
Comment (SvenRoederer) [inline README.md:31]: "with lists of packages in **packaglists/** for the router.

File: README.md [hunk 27]
```diff
@@ -1,28 +1,36 @@
...
+All additional packages can be found at http://github.com/freifunk-berlin/packages_berlin.
```
Comment (SvenRoederer) [inline README.md:32]: should we mention "gluon-packages" and "freifunk-potsdam-packages" also. (not shure if ffp-packages made it up to this repo already)

File: README.md [hunk 28]
```diff
@@ -48,6 +56,41 @@ pacman -S base-devel git ncurses lib32-zlib gawk time unzip perl-xml-libxml \
...
 
+
+### Build-System-Structure  
+
+Where can I change something? What does these files do? 
+
+```
+:: Settings
+
+config.mk             - OpenWrt git repository, revision, select what package lists to build
```
Comment (SvenRoederer) [inline README.md:67]: "OpenWrt git repository, revision" is controlled by modules-file

File: README.md [hunk 29]
```diff
@@ -48,6 +56,41 @@ pacman -S base-devel git ncurses lib32-zlib gawk time unzip perl-xml-libxml \
...
+feeds.conf            - OpenWrt feeds.conf that is used for the package feeds during build
```
Comment (SvenRoederer) [inline README.md:68]: "feeds.conf" is converted to modules

File: README.md [hunk 30]
```diff
@@ -48,6 +56,41 @@ pacman -S base-devel git ncurses lib32-zlib gawk time unzip perl-xml-libxml \
...
+
+configs/                  - further configurations
```
Comment (SvenRoederer) [inline README.md:70]: "further configurations" replace by "snippets of the OpenWrt config-options (compare to call "make menuconfig" in openwrt/)" or something alike

File: README.md [hunk 31]
```diff
@@ -94,73 +138,103 @@ firmwares/
...
-These different *packages lists* are defined in `packages/`.
-See the "Features" section above for a description of the purpose of each package list.
-With the "OpenWrt-Imagebuilder" you can assemble your own image variant with your
-*packages lists* without having to compile everything yourself. The "OpenWrt-SDK" is
-the fastest way to build your own packages or programs without compiling OpenWrt itself.
-The "initrd" directory contains some initrd-images for netboot, which are required on
-some boards to initially install OpenWrt.
+As you notice there are several different image variants ("manual", "notunnel",
+etc.). 
++ These different *packages lists* are defined in `packages/`. See the
```
Comment (SvenRoederer) [inline README.md:143]: `packages/` --> `packagelists/`

Comment (SvenRoederer) [reply README.md:143]: You talk here on the "firmware" folder, right? Then this packages folder is just the storage-repo of all packages available.These packages will be included according to the flavors in packagelists/ to the image or used as source then running opkg install on the router.

File: README.md [hunk 32]
```diff
@@ -94,73 +138,103 @@ firmwares/
...
+
+All branches whose name complies to the "X.Y.Z" pattern are built and put into
+the "stable" downloads directory: [http://buildbot.berlin.freifunk.net/buildbot/stable/](http://buildbot.berlin.freifunk.net/buildbot/stable/)
 
-All branches whose name complies to the "X.Y.Z" pattern are built and put into the "stable" downloads directory:
-[http://buildbot.berlin.freifunk.net/buildbot/stable/](http://buildbot.berlin.freifunk.net/buildbot/stable/)
+All branches with names not fitting the "X.Y.Z" pattern are built and put into
+the "unstable" directory: [http://buildbot.berlin.freifunk.net/buildbot/unstable/](http://buildbot.berlin.freifunk.net/buildbot/unstable/)
+Note that in the directory there is no reference to the branch name; unstable
+builds can be identified by build number only.
```
Comment (SvenRoederer) [inline README.md:212]: referencing the VERSION.txt file? As this is the best place to look up on which branch and revision a build is based on.

File: README.md [hunk 33]
```diff
@@ -94,73 +138,103 @@ firmwares/
...
+version has its own codename. We name our releases after important female
+computer scientists, hackers, etc. For inspiration please take a look at the
+related [ticket](https://github.com/freifunk-berlin/firmware/issues/24).
 
-For a new release, create a new branch. The branch name must be a semantic version
-number. Make sure you change the semantic version number and, for major releases,
-the codename in the README and config files (./configs/*)
+For a new release, create a new branch. The branch name must be a semantic
+version number. Make sure you change the semantic version number and, for major
+releases, the codename in the README and config files (`./configs/`)
```
Comment (SvenRoederer) [inline README.md:224]: (`./configs/`) --> in all other places we did not prefix the path by "./"

**Review by SvenRoederer — CHANGES_REQUESTED** [review]
So far thanks for your work

**Comment by Akira25** [comment]
@SvenRoederer There were 37 (±2) requests for changes in total. Please feel free to include them into the README-File. It might be faster to do it by yourself, then writing me, what I should change. I will no longer invest effort into this. You seem to be the only one, who knows how this buildsystem works.

Sadly, the documentation will still document a different state than the current one.

**Comment by SvenRoederer** [comment]
> @SvenRoederer There were 37 (±2) requests for changes in total. Please feel free to include them into the README-File. It might be faster to do it by yourself, then writing me, what I should change. 

That's the downside of having the PR source from your private Repo-fork ... Only you have permission to commit to it. My impression was, that the most of the open changes were were most "consistency of terms" and generic comments.

> I will no longer invest effort into this. You seem to be the only one, who knows how this buildsystem works.

I got used to it and found some smart ideas inside. Sadly I seems to be the only one still working on it, as the "old" developers (@sarumpaet @booo @andrenarchy @lynxis ) seem to have retired. 
> 
> Sadly, the documentation will still document a different state than the current one.

Indeed your updated documentation somehow based on the outdated system before switching to "modules"-file.

But thanks for your work so far. I'll reopen, in case someone else likes to continue here.
